src/golang/lib/engine/authenticate.go

```diff
@@ -43,11 +49,69 @@ func AuthenticateLambdaConfig(ctx context.Context, authConf auth.Config) error {
 		lambda_utils.SnowflakeConnectorType,
 	}
 
-	for _, functionType := range functionsToShip {
-		err := lambda_utils.CreateLambdaFunction(functionType, lambdaConf.RoleArn)
-		if err != nil {
-			return errors.Wrap(err, "Unable to Create Lambda Function")
-		}
+	// Run authentication only once since the credentials will be recorded.
+	err = lambda_utils.AuthenticateDockerToECR()
+	if err != nil {
+		return errors.Wrap(err, "Unable to authenticate Lambda Function.")
+	}
+
```

Comment:
Can we move the new pushing/pulling code to `lambda/utils.go`? This file is shared by a lot of other integrations.

Reply:
Sounds good! Thanks for pushing this through. I will do another around of testing after that.